libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -920,6 +1007,36 @@ export interface ISimulatorResourceInstance {
    * Save the resource's state into the state directory.
    */
   save(statedir: string): Promise<void>;
+
+  /**
+   * Determines the update plan for applying a new configuration for this resource.
+   *
+   * If this is not implemented, the default behavior is to automatically replace the resource if
+   * the new configuration is different from the current configuration.
+   *
+   * @param newConfig The new configuration to apply
+   */
+  plan(newConfig: BaseResourceSchema): Promise<UpdatePlan>;
```

Comment:
Is the resource receiving a piece of config with all tokens resolved, or is it possible there are unresolved tokens in it? Which do you think should be the contract here?

Reply:
Tokens are unresolved, which I think is reasonable. Added to the docs.